### PR TITLE
Fix/kill controller removes k8s ns

### PR DESCRIPTION
**Pull request by ycliuhw**
## Description of change

Fix: kill controller cmd now destroys k8s namespaces for caas models from cli;

## QA steps

- bootstrap

```bash
$ juju bootstrap microk8s $CNAME
```

- add-model

```bash
$ juju add-model $MNAME microk8s
$ juju deploy cs:~juju/mariadb-k8s-0
```

- kill controller

```bash
$ juju kill-controller -t 0 -y $CNAME
```

- check if the namespace has been removed

```bash
$ kubectl get ns | grep $MNAME
```

## Documentation changes

No

## Bug reference

https://bugs.launchpad.net/juju/+bug/1832669

### CODE REVIEW

File: caas/kubernetes/provider/k8s.go [hunk 1]
```diff
@@ -376,7 +375,13 @@ func (*kubernetesClient) Provider() caas.ContainerEnvironProvider {
 }
 
 // Destroy is part of the Broker interface.
-func (k *kubernetesClient) Destroy(callbacks context.ProviderCallContext) error {
+func (k *kubernetesClient) Destroy(callbacks context.ProviderCallContext) (err error) {
+	defer func() {
+		if k8serrors.ReasonForError(err) == v1.StatusReasonUnknown {
+			logger.Warningf("k8s cluster is not accessible %v", err)
```
Comment (wallyworld) [inline caas/kubernetes/provider/k8s.go]: use ":" before %v

... not accessible: %v

**Comment by wallyworld** [comment]
[kill k8s controller no longer removes model namespaces](https://trello.com/c/1rH9ENCJ/790-kill-k8s-controller-no-longer-removes-model-namespaces)

**Comment by ycliuhw** [comment]
16:13:43 
[LOG] 0:00.075 WARNING juju.testing failed to start mongo: mongod failed to listen on port 44920

!!build!!

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**